### PR TITLE
Allow setting an SSO provider via env variables

### PR DESCRIPTION
**Pull request by pgoetze**
We need the provider to be configurable for our different environments (Dev should use the dev SSO, staging should use the staging SSO, etc.)

The provider configured in the env vars and the ones from `config.js` are merged.
Only providers with non-empty text and url are used.

The env vars for setting the SSO provider are:
- `SSO_PROVIDER_TEXT`
- `SSO_PROVIDER_URL`

ping @eschaefer, @mattberridge, @Tom-Bonnike for review

(I have no idea how to test this helper with all the internal stuff being loaded. I guess this means it’s poorly designed 😕 , any ideas how to improve it?)

### CODE REVIEW

**Comment by mattberridge** [comment]
can you define test `ENV` values in `webpack.config.test.js`?

**Comment by mattberridge** [comment]
but yeah it's not that testable. unless you pass `settings` in as an argument to the function to make it closer to being 'pure'?

**Comment by pgoetze** [comment]
Jep, defining test `ENV` values should help to test it against the env provider. I would like passing the ssoProviders as a param! I think I will change it.

**Comment by pgoetze** [comment]
@mattberridge ssoProviders are now passed as argument.

File: test/helpers/generateSsoProviders.js [hunk 1]
```diff
@@ -7,15 +7,14 @@ describe('generateSsoProviders', () => {
   const returnPath = '/return-url';
   const escapedReturnUrl = escape([envVars.baseUrl, returnPath].join(''));
 
+  const ssoProvidersFromEnv = envVars.ssoProviderText && envVars.ssoProviderUrl
```
Comment (mattberridge) [inline test/helpers/generateSsoProviders.js]: @pgoetze I added this so tests won't fail when you haven't defined any providers in `ENV`.

**Pull request merged**